modules.js

```diff
@@ -52,6 +51,23 @@ var self = module.exports = {
       return false
     }
   },
+  unloadModule: function (bot, file) {
+    // check if module already loaded:
+    const filename = require.resolve(file)
+
+    if (filename in require.cache) {
+      const alreadyLoadedModule = require(file)
+      if (alreadyLoadedModule.onunload) {
+        alreadyLoadedModule.onunload(bot)
```

Comment:
If this throws, it's impossible to unload the module. Is it better to cancel the unload or to forcibly remove the module anyway?